### PR TITLE
Deploy top-of-tree xbuildenv and pyodide-core to S3

### PR DESCRIPTION
**Pull request by ryanking13**
Resolve: #3759

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -556,6 +557,19 @@ jobs:
...
+      - run:
+          # Unlike the release version, we upload the dev version to S3 not to GitHub.
+          name: Deploy release files to S3
+          command: |
+            mkdir -p /tmp/ghr/dist
+            cp -r dist /tmp/ghr/pyodide
+            cp -r xbuildenv /tmp/ghr/xbuildenv
+            cd /tmp/ghr
+            tar cjf dist/pyodide-core.tar.bz2 pyodide/pyodide{.js,.mjs,.asm.js,.asm.wasm} pyodide/{package,repodata}.json pyodide/python_stdlib.zip
+            tar cjf dist/xbuildenv.tar.bz2  xbuildenv/
```
Comment (rth) [inline .circleci/config.yml:569]: Ahh, I think we haven't whitelisted this extension with jsdelivr so it won't work. 

Maybe we could put it to `pyodide-cache` bucket where it was originally,
```
python3 tools/deploy_s3.py dist/ "xbuildenv/" --bucket "pyodide-cache" --cache-control 'max-age=3600, public' --rm-remote-prefix
```
but then we need to pass it different credentials somehow. For instance maybe define `AWS_ACCESS_KEY_ID_CACHE`,  `AWS_SECRET_ACCESS_KEY_CACHE` (sent you by email) and then set them back to the variables without the `_CACHE` prefix just for the below command.

Comment (ryanking13) [reply .circleci/config.yml:569]: Thanks, I changed the bucket and added the corresponding keys.

**Review by rth — APPROVED** [review]
Thanks!

**Comment by rth** [comment]
Just a couple of test failures. 
```
FAILED tools/tests/test_deploy_s3.py::deploy_to_s3_overwrite - TypeError: str expected, not OptionInfo
FAILED tools/tests/test_deploy_s3.py::deploy_to_s3_mime_type - TypeError: str expected, not OptionInfo
```
you need to explicitly pass all arguments in tests, as default typer arguments are then not valid.

**Pull request merged**